pinot-core/src/main/java/org/apache/pinot/core/query/postaggregation/PostAggregationFunction.java

```diff
@@ -38,20 +37,20 @@ public class PostAggregationFunction {
 
   public PostAggregationFunction(String functionName, ColumnDataType[] argumentTypes) {
     String canonicalName = FunctionRegistry.canonicalize(functionName);
-    FunctionInfo functionInfo = FunctionRegistry.lookupFunctionInfo(canonicalName, argumentTypes);
+    int numArguments = argumentTypes.length;
+    FunctionInfo functionInfo = FunctionRegistry.lookupFunctionInfo(canonicalName, numArguments);
     if (functionInfo == null) {
       if (FunctionRegistry.contains(canonicalName)) {
         throw new IllegalArgumentException(
-            String.format("Unsupported function: %s with argument types: %s", functionName,
-                Arrays.toString(argumentTypes)));
+            String.format("Unsupported function: %s with %d parameters", functionName, numArguments));
```

Comment:
I would recommend to do not use `String.format` if not needed. In this case it is easier (and cheaper!) to just concatenate text. It is different if `Preconditions` is used, given there we would need to first call toString on arguments,t hen concatenate... just to verify then verify that the String will not be used.

Reply:
Forget about this. I didn't see the older code was already using this (IMHO worse) option